### PR TITLE
Support setting time offset on test server within tests

### LINKED ISSUE #34189: Support setting time offset on test server within tests
Quite a few tests rely on setting the time offset on the server, and this should be a core capability of the framework.

An example of how this should be done by a test:

```
MyTest {
   @InjectTimeOffset(offset = 10l, lifecycle = LifeCycle.CLASS)
   TimeOffset timeOffset;

   @Test
   public void someTest() {
      timeOffset.set(20l);
      timeOffset.get();
   }

  @Test
  public void anotherTest() {}
}
```

The above makes it possible for a test to set a default time-offset for all tests, as well as override it per-test, or within steps in the test.

The offset should be reset to the original value, so in the example above after `someTest` the value should be reset to `10l`, and after the test class completes it should be reset to 0.

Behind the scenes initially this should rely on the [RunOnSever](https://github.com/keycloak/keycloak/issues/34190) capability.

### PR DESCRIPTION
**Pull request by vaceksimon**
Closes: #34189

### CODE REVIEW

File: test-framework/remote/src/main/java/org/keycloak/test/framework/remote/timeoffset/TimeOffsetSupplier.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    @Override
+    public int order() {
+        return SupplierOrder.BEFORE_KEYCLOAK_SERVER;
+        // Implementing the KeycloakServerConfigInterceptor is a workaround for RemoteProvidersSupplier to work
+    }
+
+    @Override
+    public KeycloakServerConfigBuilder intercept(KeycloakServerConfigBuilder serverConfig, InstanceContext<TimeOffSet, InjectTimeOffSet> instanceContext) {
+        return serverConfig;
+    }
```
Comment (stianst) [inline test-framework/remote/src/main/java/org/keycloak/test/framework/remote/timeoffset/TimeOffsetSupplier.java:64]: Strange that this is needed, or that it works; might need to tweak this a little bit so we can allow dependencies to be server config interceptors as well. We might need to add something like getDependencies or something to suppliers :/

File: test-framework/remote/pom.xml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>keycloak-test-framework-remote</artifactId>
+    <name>Keycloak Test Framework - Remote testing utils</name>
+    <packaging>jar</packaging>
+    <description>Utility Keycloak server Suppliers for the Keycloak Test Framework</description>
+
+    <dependencies>
+        <dependency>
```
Comment (stianst) [inline test-framework/remote/pom.xml:36]: You need to add a dependency here on the remote-providers, as that will in turn make any test module that depends on this to have a dependency on the remote-providers. That's probably the last piece to the puzzle to get this one to pass on GA

**Review by stianst — COMMENTED** [review]
Overall LGTM. We can merge once it completes GA, and probably handle the issue around interceptors as a follow-up. I can take a look at that if you want, I suspect we may end up having to have some explicit declaration of dependencies in suppliers though :/

File: test-framework/remote/pom.xml (hunk 2 above)
Comment (vaceksimon) [reply test-framework/remote/pom.xml:36]: Yes, I resolved the pom dependency locally. It works running with maven (`mvn install` and `mvn test`)

**Review by stianst — APPROVED** [review]

**Pull request merged**